### PR TITLE
GS and physics small fixes

### PR DESCRIPTION
**Pull request by CedricGuillemet**
Fixes for : 
- https://forum.babylonjs.com/t/havok-plugin-dispose-returns-error/50087
- https://forum.babylonjs.com/t/gaussian-splatting-does-not-support-right-handed-system/50260
- https://forum.babylonjs.com/t/havok-incorrect-scaling-of-mesh-when-dragging-with-constraint/50113

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15097/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15097/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15097/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15097/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15097/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15097/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15097/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15097/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15097/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)
Important - these might fail sporadically. This is an optional test.

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15097/merge/testResults/webgpuplaywright/index.html

File: packages/dev/core/src/Physics/v2/Plugins/havokPlugin.ts [hunk 1]
```diff
@@ -288,7 +288,7 @@ export class HavokPlugin implements IPhysicsEnginePluginV2 {
     /**
      * We only have a single raycast in-flight right now
      */
-    private _queryCollector: bigint;
+    private _queryCollector;
```
Comment (ryantrem) [inline packages/dev/core/src/Physics/v2/Plugins/havokPlugin.ts:291]: Why was the type removed here?

Comment (CedricGuillemet) [reply packages/dev/core/src/Physics/v2/Plugins/havokPlugin.ts:291]: queryCollector is not a `bigint` but a `[bigint]`. It's actually an opaque type that has no meaning outside Havok API calls.

File: packages/dev/core/src/Physics/v2/Plugins/havokPlugin.ts [hunk 2]
```diff
@@ -2267,10 +2271,14 @@ export class HavokPlugin implements IPhysicsEnginePluginV2 {
      * Dispose the world and free resources
      */
     public dispose(): void {
-        this._hknp.HP_QueryCollector_Release(this._queryCollector);
-        this._queryCollector = BigInt(0);
-        this._hknp.HP_World_Release(this.world);
-        this.world = undefined;
+        if (this._queryCollector) {
+            this._hknp.HP_QueryCollector_Release(this._queryCollector);
+            this._queryCollector = BigInt(0);
```
Comment (RaananW) [inline packages/dev/core/src/Physics/v2/Plugins/havokPlugin.ts:2276]: if this is the incorrect type, why not set it to undefined?

**Review by Popov72 — APPROVED** [review]

File: packages/dev/core/src/Physics/v2/Plugins/havokPlugin.ts (hunk 2 above)
Comment (CedricGuillemet) [reply packages/dev/core/src/Physics/v2/Plugins/havokPlugin.ts]: Replaced with `undefined`

**Pull request merged**